src/sdkClient/__tests__/clientAttributesDecoration.spec.ts

```diff
@@ -0,0 +1,270 @@
+import { sdkClientMethodCSFactory } from '../sdkClientMethodCS';
+
+import { settingsWithKey } from '../../utils/settingsValidation/__tests__/settings.mocks';
+
+const partialStorages: { destroy: jest.Mock }[] = [];
+
+const storageMock = {
+  destroy: jest.fn(),
+  shared: jest.fn(() => {
+    partialStorages.push({ destroy: jest.fn() });
+    return partialStorages[partialStorages.length - 1];
+  })
+};
+
+const partialSdkReadinessManagers: { sdkStatus: jest.Mock, readinessManager: { destroy: jest.Mock } }[] = [];
+
+const sdkReadinessManagerMock = {
+  sdkStatus: jest.fn(),
+  readinessManager: {
+    destroy: jest.fn(),
+    isDestroyed: jest.fn(() => { return false; }),
+    isReady: jest.fn(() => { return true; })
+  },
+  shared: jest.fn(() => {
+    partialSdkReadinessManagers.push({
+      sdkStatus: jest.fn(),
+      readinessManager: { destroy: jest.fn() },
+    });
+    return partialSdkReadinessManagers[partialSdkReadinessManagers.length - 1];
+  })
+};
+
+const partialSyncManagers: { start: jest.Mock, stop: jest.Mock, flush: jest.Mock }[] = [];
+
+const syncManagerMock = {
+  stop: jest.fn(),
+  flush: jest.fn(() => Promise.resolve()),
+  shared: jest.fn(() => {
+    partialSyncManagers.push({ start: jest.fn(), stop: jest.fn(), flush: jest.fn(() => Promise.resolve()) });
+    return partialSyncManagers[partialSyncManagers.length - 1];
+  })
+};
+
+const params = {
+  storage: storageMock,
+  sdkReadinessManager: sdkReadinessManagerMock,
+  syncManager: syncManagerMock,
+  signalListener: { stop: jest.fn() },
+  settings: settingsWithKey
+};
+
+// We are testing attributes binding feature and we just need to know how the attributes are combined before evaluation
+// So input validation decorator is mocked to return the combined attributes instead of evaluation so we can verify them
+jest.mock('../clientInputValidation', () => {
+  return {
+    clientInputValidationDecorator() {
+      return {
+        getTreatment(maybeKey: any, maybeSplit: string, maybeAttributes?: any) {
+          return maybeAttributes;
+        },
+        getTreatmentWithConfig(maybeKey: any, maybeSplit: string, maybeAttributes?: any) {
+          return maybeAttributes;
+        },
+        getTreatments(maybeKey: any, maybeSplits: string[], maybeAttributes?: any) {
+          return maybeAttributes;
+        },
+        getTreatmentsWithConfig(maybeKey: any, maybeSplits: string[], maybeAttributes?: any) {
+          return maybeAttributes;
+        }
+      };
+    }
+  };
+});
+
+// @ts-expect-error
+const sdkClientMethod = sdkClientMethodCSFactory(params);
+const client = sdkClientMethod();
+
+//expect(client).toBe(AttributesDecorationMockedClient);
```

Comment:
Since it is a UT for `clientAttributesDecoration`, it is better to test it calling the `clientAttributesDecoration` function directly. 
So, replace all lines above, with something like the following:

```
import { clientAttributesDecoration } from '../clientAttributesDecoration';
import { loggerMock } from '../../logger/__tests__/sdkLogger.mock';

// mocked methods return the provided attributes object (2nd argument), to assert that it was properly passed
const clientMock = {
  getTreatment(maybeKey: any, maybeSplit: string, maybeAttributes?: any) {
    return maybeAttributes;
  },
  getTreatmentWithConfig(maybeKey: any, maybeSplit: string, maybeAttributes?: any) {
    return maybeAttributes;
  },
  getTreatments(maybeKey: any, maybeSplits: string[], maybeAttributes?: any) {
    return maybeAttributes;
  },
  getTreatmentsWithConfig(maybeKey: any, maybeSplits: string[], maybeAttributes?: any) {
    return maybeAttributes;
  }
};
// @ts-expect-error
const client = clientAttributesDecoration(loggerMock, clientMock);
```